### PR TITLE
[SPARK-36773][SQL][TEST] Fixed unit test to check the compression for parquet

### PR DESCRIPTION
**Pull request by klaus-xiong**
### What changes were proposed in this pull request?
Change the unit test for parquet compression

### Why are the changes needed?
To check the compression for parquet

### Does this PR introduce _any_ user-facing change?
no 

### How was this patch tested?
change unit test

### CODE REVIEW

**Comment by BelodengKlaus** [comment]
@wzhfy

**Comment by sunchao** [comment]
what is "uts"? do you mean unit tests?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala [hunk 1]
```diff
@@ -105,7 +105,7 @@ class ParquetCompressionCodecPrecedenceSuite extends ParquetTest with SharedSpar
 
   test("Create parquet table with compression") {
     Seq(true, false).foreach { isPartitioned =>
-      Seq("UNCOMPRESSED", "SNAPPY", "GZIP").foreach { compressionCodec =>
+      Seq("UNCOMPRESSED", "SNAPPY", "GZIP", "LZ4", "ZSTD").foreach { compressionCodec =>
```
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: `org.apache.hadoop.io.compress.ZStandardCodec` is added at Apache Hadoop 2.9.0+. Does the test pass with Hadoop 2 profile?

Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: cc @viirya too

Comment (sunchao) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: Parquet uses its own ZSTD codec, see https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/codec/ZstandardCodec.java

Comment (HyukjinKwon) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: 👌

**Comment by HyukjinKwon** [comment]
ok to test

**Review by HyukjinKwon — APPROVED** [review]

**Comment by BelodengKlaus** [comment]
> what is "uts"? do you mean unit tests?

yes, it means unit tests

**Comment by HyukjinKwon** [comment]
^^ can you fix PR title and description ut -> unit test?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala (hunk 1 above)
Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: yea

**Comment by viirya** [comment]
+1 for replacing "uts" with "unit tests".

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala (hunk 1 above)
Comment (BelodengKlaus) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetCompressionCodecPrecedenceSuite.scala:108]: I have test in my own env with the "LZ4", "ZSTD" and "BROTLI". The "BROTLI" is a hadoop build in plugin, we can not check this compression, and the "ZSTD" and "LZ4" import jar packages to support those compression. So I think we need to check those compression can work well when we upgrade the parquet , zstd and lz4 version.

**Comment by BelodengKlaus** [comment]
> ^^ can you fix PR title and description ut -> unit test?

changed, thanks for the advice

**Comment by BelodengKlaus** [comment]
> +1 for replacing "uts" with "unit tests". it is pretty confusing.
Thanks for the change of title. ^ ^

**Review by sunchao — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by BelodengKlaus** [comment]
Thanks for review

**Pull request closed without merging**